src/OpenSearch/Aws/SigningRequestFactory.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace OpenSearch\Aws;
+
+use Aws\Credentials\CredentialsInterface;
+use Aws\Signature\SignatureInterface;
+use OpenSearch\RequestFactory;
+use OpenSearch\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * A decorator request factory that signs requests using the provided AWS credentials and signer.
+ */
+class SigningRequestFactory implements RequestFactoryInterface
```

Comment:
This would be better implemented as a client decorator than as a request factory decorator. signing must happen *after* any transformation applied on the Request